databricks/koalas/base.py

```diff
@@ -220,7 +234,24 @@ def __sub__(self, other):
                 raise TypeError("date subtraction can only be applied to date series.")
         return column_op(Column.__sub__)(self, other)
 
-    __mul__ = column_op(Column.__mul__)
+    def __mul__(self, other):
+        if isinstance(self, str) or isinstance(other, str):
+            raise TypeError("multiplication can not be applied to a string literal")
+
+        if (
+            isinstance(self.spark.data_type, NumericType)
```

Comment:
`IntegralType` instead of `NumericType`?

```py
>>> df = pd.DataFrame({'a':[1.5, 2.5, 3.5], 'b':['a', '2', 'c']})
>>> df
     a  b
0  1.5  a
1  2.5  2
2  3.5  c
>>> df.a * df.b
Traceback (most recent call last):
...
TypeError: can't multiply sequence by non-int of type 'float'
```

Reply:
Good idea! This also makes changes in this PR consistent (between int and str columns).